designs/expose-metrics.md

```diff
@@ -0,0 +1,371 @@
+# Expose metrics for all EKS-A components securely
+
+## Problem Statement
+
+Customers want to scrape the metrics of various EKS-A components with prometheus in order to understand and monitor the state of a cluster. In EKS-A, metrics of only some k8s system components (kube-apiserver, kubelet, coredns, kube-vip, cert-manager, cilium) are exposed by default. Other system components such as `kube-controller-manager` and `kube-scheduler` are configured with the default `--bind-address=127.0.0.1` (localhost). 
+
+Below are some examples of customer requests for exposing metrics:
+
+* https://github.com/aws/eks-anywhere/issues/4299
+* https://github.com/aws/eks-anywhere/issues/4405
+* https://github.com/aws/eks-anywhere/issues/7106
+
+## Goals and Objectives
+
+As an EKS-A user, I would like to:
+
+* expose all k8s system component metrics securely with authentication and authorization enabled
+* expose metrics from all EKS-A, CAPI etcd components and CAPI provider-specific components securely
+
+## Statement of Scope
+
+**In Scope:**
+
+Exposing metrics securely for the following components:
+
+1. k8s system components
+
+* kube-controller-manager
+* kube-scheduler
+* kube-proxy
+
+2. EKS-A components
+
+* eks-anywhere-controller-manager controller
+* eks-anywhere-packages controller
+
+3. CAPI etcd components
+
+* etcdadm-bootstrap-provider controller
+* etcdadm-controller-controller-manager controller
+
+4. CAPI provider-specific components
+
+* capt-controller-manager controller
+* capc-controller-manager controller
+* capv-controller-manager controller
+* capx-controller-manager controller
+
+**Out of Scope:**
+
+Following components are not considered for exposing metrics securely:
+
+* Snow provider (capas) and docker provider (capd)
+* ECR-credential-provider
+
+**Future Scope:**
+
+* Securely expose metrics for all other components (kube-vip, coredns, cilium, cert-manager)
+
+## Current State of EKS-A components
+
+![table](images/expose-metrics.png)
```

Comment:
nit: can we make this a md table instead of an image? updating the image is a pain

Reply:
Agree with that but then in this case, the md table won't be properly formatted in the way it looks in the image. Considering that I would prefer keeping the image as we won't be updating the image frequently.